src/trio/_tests/test_socket.py

```diff
@@ -629,6 +629,9 @@ async def res(
                 # Linux
                 if hasattr(tsocket, "EAI_ADDRFAMILY"):
                     expected_errnos.add(tsocket.EAI_ADDRFAMILY)
+                # musl libc
+                if hasattr(tsocket, "EAI_NODATA"):
+                    expected_errnos.add(tsocket.EAI_NODATA)
```

Comment:
codecov says this comparison never fails, so we should either add it unconditionally (but some quick searching turns up https://github.com/python/cpython/issues/69325) or add `# pragma: no branch`

Reply:
I think it's fine to have tests fail then add in the branch at that time